lib/event_store_http_client/events/write.rb

```diff
@@ -14,14 +14,12 @@ class Write
         dependency :settings
         dependency :client
 
-        def initialize(type, data, stream_name, version)
```

Comment:
But it would know that it's an "account" stream, where account is effectively a category, and the account "account-621" stream name is "#{category}-#{id}".

The closer these abstractions come to concrete, scenario code, the more concrete they should become. I would suspect that we'd get to the point where the scenario code has something like:

```ruby
account_stream = AccountStream.build ...
account_stream.write account_id, type, data, version
```